### PR TITLE
WIP: test code hash

### PR DESCRIPTION
**Pull request by fulldecent**
# Blockers 

- [x] https://github.com/0xcert/ethereum-erc721/issues/246

### CODE REVIEW

File: src/tests/tokens/nf-token.test.ts [hunk 1]
```diff
@@ -397,4 +397,14 @@ spec.test('throws when trying to burn non existant NFT', async (ctx) => {
...
+});
+
+spec.test('safeTransfer does not call onERC721Received to constructing contract', async (ctx) => {
+  const sendsToSelfOnConstruct = await ctx.deploy({ 
+    src: './build/sends-to-self-on-construct.json',
+    contract: 'SendsToSelfOnConstruct',
+  });
+
+  const logs = sendsToSelfOnConstruct.logs;
+  ctx.is(logs.events.Received, undefined);
```
Comment (MoMannn) [inline src/tests/tokens/nf-token.test.ts]: Deploy is a bit different then other transactions. It returns an object with an `instance` which is an instance of the deployed contract and `receipt` with information about the deployment. Receipt is what you should look at. If receipt does not contain what you need try updating to the latest specron version: 0.16.0

**Comment by fulldecent** [comment]
@MoMannn Updated to latest, using receipt. Have one little issue with testing the event.

**Comment by xpepermint** [comment]
@fulldecent What kind of issue?

File: src/tests/tokens/nf-token.test.ts [hunk 2]
```diff
@@ -397,4 +397,15 @@ spec.test('throws when trying to burn non existant NFT', async (ctx) => {
...
+
+spec.test('safeTransfer does not call onERC721Received to constructing contract', async (ctx) => {
+  const sendsToSelfOnConstruct = await ctx.deploy({ 
+    src: './build/sends-to-self-on-construct.json',
+    contract: 'SendsToSelfOnConstruct',
+  });
+
+  const receipt = sendsToSelfOnConstruct.receipt;
+  console.log(receipt.events.Received());
+  ctx.not(receipt.events.Received, undefined); // I want to confirm here that there is only one event (the mint(), and not the safeTransferFrom)
```
Comment (MoMannn) [inline src/tests/tokens/nf-token.test.ts]: You have all the emmited events here. So even if you are expecting same kind of event. Lets say `Transfer` you will have an array of `Transfer` events here and can check if there is only one and how it is constructed.

**Comment by xpepermint** [comment]
@fulldecent what should we do with this?

**Comment by fulldecent** [comment]
We should fix this up and merge. Can bill the effort to our client because 721 is being used on that other blockchain.

And this PR is useful for blockchains other than Ethereum Mainnet.

**Comment by xpepermint** [comment]
@fulldecent status?

**Comment by MoMannn** [comment]
@fulldecent 
What is the point of this? You are only adding a test where you send yourself a token? What is the expected result/meaning for this?

**Comment by fulldecent** [comment]
This test is checking whether a token contract, which sends itself a token during contract creation, will trigger `onERC721Received`.

This behavior is expected to be different based on using EXTCODESIZE and EXTCODEHASH (pre- and post- Homestead versions).

The implementation here supports Ethereum (EXTCODEHASH) but the implementation in 0xcert Framework supports pre-Homestead (notably, other chains pre Homestead) so I wanted to make a test that illustrates this difference.

**Comment by MoMannn** [comment]
@fulldecent I fixed the check so it verifies there is no Received event emitted. There are only 2 Transfer events (first for creating the token, the second for sending the token to yourself.)

**Review by MoMannn — APPROVED** [review]

**Comment by xpepermint** [comment]
@fulldecent please verify.

**Comment by fulldecent** [comment]
Thank you, LGTM

**Pull request merged**